src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -229,6 +229,14 @@ void Simulator::handle_message(mavlink_message_t *msg, bool publish) {
 		update_sensors(&imu);
 		break;
 
+	case MAVLINK_MSG_ID_HIL_OPTICAL_FLOW:
+		mavlink_hil_optical_flow_t flow;
+		mavlink_msg_hil_optical_flow_decode(msg, &flow);
+		if (publish) {
+			publish_flow_topic(&flow);
```

Comment:
You should not check the publish flag here, because it won't work else in default mode. Just publish always.